bundles/org.openhab.binding.doorbird/src/main/java/org/openhab/binding/doorbird/internal/api/DoorbirdAPI.java

```diff
@@ -0,0 +1,255 @@
+        return downloadImage("/bha-api/history.cgi?event=motionsensor&index=" + imageNumber);
+    }
+
+    public void openDoorController(String controllerId, String doorNumber) {
+        openDoor("/bha-api/open-door.cgi?r=" + controllerId + "@" + doorNumber);
+    }
+
+    public void openDoorDoorbell(String doorNumber) {
+        openDoor("/bha-api/open-door.cgi?r=" + doorNumber);
+    }
+
+    private void openDoor(String urlFragment) {
+        try {
+            String response = executeGetRequest(urlFragment);
+            logger.debug("Response={}", response);
+        } catch (IOException e) {
+            logger.debug("IOException opening door: {}", e.getMessage());
+        } catch (DoorbirdUnauthorizedException e) {
+            logAuthorizationError("openDoor");
+        }
+    }
+
+    private @Nullable synchronized DoorbirdImage downloadImage(String urlFragment) {
+        Authorization auth = authorization;
+        if (auth == null) {
+            logAuthorizationError("downloadImage");
+            return null;
+        }
+        HttpClient client = httpClient;
+        if (client == null) {
+            logger.info("Unable to download image because httpClient is not set");
+            return null;
+        }
+        String errorMsg;
+        try {
+            String url = buildUrl(auth, urlFragment);
+            logger.debug("Downloading image from doorbird: {}", url);
+            Request request = client.newRequest(url);
+            request.method(HttpMethod.GET);
+            request.header("Authorization", "Basic " + auth.getAuthorization());
+            request.timeout(API_REQUEST_TIMEOUT_SECONDS, TimeUnit.SECONDS);
+
+            ContentResponse contentResponse = request.send();
+            switch (contentResponse.getStatus()) {
+                case HttpStatus.OK_200:
+                    DoorbirdImage doorbirdImage = new DoorbirdImage();
+                    doorbirdImage.setImage(new RawType(contentResponse.getContent(),
+                            contentResponse.getHeaders().get(HttpHeader.CONTENT_TYPE)));
+                    doorbirdImage.setTimestamp(convertXTimestamp(contentResponse.getHeaders().get("X-Timestamp")));
+                    return doorbirdImage;
+
+                default:
+                    errorMsg = String.format("HTTP GET failed: %d, %s", contentResponse.getStatus(),
+                            contentResponse.getReason());
+                    break;
+            }
+        } catch (TimeoutException e) {
+            errorMsg = "TimeoutException: Call to Doorbird API timed out";
+        } catch (ExecutionException e) {
+            errorMsg = String.format("ExecutionException: %s", e.getMessage());
+        } catch (InterruptedException e) {
+            errorMsg = String.format("InterruptedException: %s", e.getMessage());
+            Thread.currentThread().interrupt();
+        }
+        logger.debug("{}", errorMsg);
```

Comment:
```suggestion
        logger.debug(errorMsg);
```

Reply:
I had it that way originally, but SAT complained about it. I switched it to `logger.debug("{}", errorMsg);` to make SAT happy.